src/http_protocol.cpp

```diff
@@ -31,6 +33,27 @@ std::string url_decode(const std::string& encoded) {
   return decoded_str;
 }
 
+bool is_method_allowed(uint8_t mask, method_t method) {
+  return mask & static_cast<uint8_t>(method);
+}
```

Comment:
i would just remove this entirely and do the bitwise operation in the 2 places its called. the function is too trivial to exist

```suggestion
```